typescript/src/game_loop/game-rxjs.ts

```diff
@@ -1,115 +1,158 @@
-import { mergeMap, Observable, Subscriber, Subscription, tap } from "rxjs";
-import { onErrorResumeNext } from "rxjs/operators";
-import { createLoserMessage, createMessage, createWinnerMessage, errorGameOver, MessageType } from "../message";
+import { mergeMap, Observable, Subscriber } from "rxjs";
+import {
+  createLoserMessage,
+  createMessage,
+  createReadyUpMessage,
+  createWinnerMessage,
+  errorGameOver,
+  Message,
+  MessageType,
+} from "../message";
 import { Server } from "../server/rxjs-server";
 import { BaseSocket, RxSocket } from "../server/universal-types";
 import { GameStat } from "../stats";
-import { GameLoopRxJS } from "./game-loop-timer";
-import { GameQueueRxJSImpl } from "./game-queue";
-import { setupWithRxJS } from "./game-setup";
+import { getRxJSGameLoop } from "./game-loop-timer";
 import GameWorld from "./game-world";
 
 function getTickRate(): number {
-    if (!process.env.TICK_RATE) {
-        return 60;
-    }
-    return +process.env.TICK_RATE || 60;
+  if (!process.env.TICK_RATE) {
+    return 60;
+  }
+  return +process.env.TICK_RATE || 60;
+}
+export interface GameResults {
+  readonly stats: GameStat;
+  readonly winner: BaseSocket;
+  readonly loser: BaseSocket;
+  readonly disconnected: boolean;
 }
 
-export type GameResults = [GameStat, BaseSocket, BaseSocket];
-export function runRxJSLoop([s1, s2]: [RxSocket, RxSocket]): Observable<GameResults> {
-    return new Observable((observer: Subscriber<GameResults>) => {
-        const subs: Subscription[] = [];
-        const stats = new GameStat();
-        const queue = new GameQueueRxJSImpl(s1, s2);
-        const world = new GameWorld(s1, s2);
-        const loop = new GameLoopRxJS(getTickRate());
-
-        function close(other: RxSocket) {
-            if (world.done) {
-                return;
-            }
-
-            loop.stop();
-            world.stop();
-            other.push(errorGameOver("The other player disconnected"));
-            observer.error(new Error("Disconnected"));
-        }
+class GameResultsImpl implements GameResults {
```

Comment:
Small change for V8 optimization. Just making sure that everything passed around here is "same shaped".